core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -249,6 +249,7 @@ abstract class SparkFunSuite
       if (level.isDefined) {
         logger.setLevel(level.get)
         logger.get().setLevel(level.get)
+        LogManager.getContext(false).asInstanceOf[LoggerContext].updateLoggers()
```

Comment:
@viirya, I borrowed your fix from https://github.com/apache/spark/pull/35012 but TBH I have less background here. Do you mind reviewing this please? This change fixes the issue in `withLogAppender` where the logger level is not changed.

Reply:
I think the issue is the same. Once the log level is set, it doesn't get changed. That's why it passed in IDE when I run it individually, and it fails when I run them in batch via SBT.

Reply:
I don't mind making this in a separate PR if you guys feel that's better.

Reply:
lgtm. I thought we should update the loggers here too. I actually was wondering if this should be do the same, but as no test failure happens I thought it maybe fine. :)

Reply:
Thanks!